packages/delisp-core/__tests__/infer.ts

```diff
@@ -169,5 +175,20 @@ describe("Type inference", () => {
         );
       });
     });
+
+    describe("Type aliases", () => {
+      const env = {
+        variables: {},
+        types: { ID: readType("number").mono }
```

Comment:
It's a bit weird that this is a `Monotype` but the variables in the env (the primitives) are `Type`.

Reply:
The reason for it is that (poly)type can only happen at toplevel for now. So if you have something like `(-> Person string)`,  if person has a "forall", it can't be replaced inside.

I can add a comment to clarify it.

But anyway, the use of Monotype vs Type everywhere is super confusing. Let's refactor that later!  And it probably makes sense to renme Monotype to Type (as it is everywhere) and Type to Polytype/Schema.